src/train_tokenizer.py

```diff
@@ -0,0 +1,142 @@
+'''Script to train tokenizers from a Hugging Face dataset'''
+
+from tokenizers import (trainers, pre_tokenizers, tokenizers, normalizers,
+                        models, AddedToken, processors, decoders)
+from datasets.arrow_dataset import Dataset
+from pathlib import Path
+from typing import Union
+from .tokenizer_config import TokenizerConfig
+
+
+def train_tokenizer(dataset: Dataset,
+                    config: Union[TokenizerConfig, dict],
+                    save_tokenizer: bool = True,
+                    output_dir: Union[str, Path] = '.',
+                    show_progress: bool = True):
+    '''Train a tokenizer on a dataset.
+
+    Args:
+        dataset (Dataset):
+            Dataset to train the tokenizer on. Must have a feature named
+            'text'.
```

Comment:
```suggestion
        texts (Iterable[str]): An iterable of strings to train the tokenizer on.
```

Reply:
Fair enough, that would make it more versatile for sure. I won't accept your suggestion here, as the unit tests will fail. For maximum versatility, I'll set it up to accept either a HF Dataset, or an iterable of texts.

Reply:
I have now changed the input to accept either a `Dataset`, an `IterableDataset` or an iterable of strings. Also included a few more unit tests to test these, and they all run successfully.

Reply:
Perfect, yea this is probably the best way to go.